jenkins/opensearch-ruby/Jenkinsfile

```diff
@@ -0,0 +1,66 @@
+pipeline {
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+                name: 'RELEASE_BRANCH',
+                description: 'Branch name to release',
+                trim: true
+        )
+    }
+    stages {
+        stage('ruby-build-sign-upload') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211130'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    if (params.RELEASE_BRANCH == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('RELEASE_BRANCH parameter is not set')
+                    }
+                    git url: 'https://github.com/opensearch-project/opensearch-ruby.git', branch: "$RELEASE_BRANCH"
+                    withCredentials([string(credentialsId: 'opensearchproject-rubygems-private-key', variable: 'RUBYGEMS_PRIVATE_KEY')]) {
+                        writeFile file: 'private_key.pem', text: RUBYGEMS_PRIVATE_KEY
+                    }
+                    sh'''
+                        keypath=$(realpath private_key.pem)
+                        for f in $(find . -name '*.gemspec');\\
+                            do echo $f;\\
+                            sed -i '$s#end#  s.signing_key = File.expand_path("'"$keypath"'") if $0 =~ /gem\\\\z/\\n&#' $f;\\
```

Comment:
So signing for ruby works little different. You have to include the above line in `.gemspec` file and then build the component.

Reply:
Doing this with groovy code would involve creating another groovy class which would be very specific to this functionality only. Also, for groovy to access file causes sandboxing access errors. Therefore, I believe it may become more complicated. Do we need this here?